custom_components/dreo/dreoheater.py

```diff
@@ -153,7 +153,6 @@ def preset_mode(self) -> str | None:
     def extra_state_attributes(self) -> dict[str, Any]:
         """Return the state attributes of the heater."""
         return {
-            "current_temperature": self.device.temperature,
```

Comment:
This additional attribute was overwriting the value being returned by the current_temperature property on line 235 and causing the display issue as HA wasn't converting it correctly.